when columns removed for an incremental model

RatandeepSharma opened the pull request:
## What are you changing in this pull request and why?
If we remove a column from an incremental model and `on_schema_change: ignore` then execute dbt run, it results in an error. However, based on the documentation, it seems that dbt run should not fail in this scenario. I find this confusing.


##Link of the page
https://docs.getdbt.com/docs/build/incremental-models#default-behavior

RatandeepSharma committed:
removed columns for an incremental model behaviour

I found confusion line as it states "Similarly, if you remove a column from your incremental model, and execute a dbt run, this column will not be removed from your target table."

But during the actual dbt run, it will encounter an error, though the documentation feels that dbt run should not fail.

Link of the page
https://docs.getdbt.com/docs/build/incremental-models#default-behavior

nghi-ly committed:
Merge branch 'current' into patch-1

nghi-ly committed:
Merge branch 'current' into patch-1

nghi-ly commented:
thanks for opening this PR, @RatandeepSharma ! appreciate your contribution to the dbt Labs docs! 

@joellabes : i think it's ok to approve and merge this change but want to double-check with you first. do you agree? or have any suggestions? thanks!

nghi-ly commented on website/docs/docs/build/incremental-models.md:
```diff
@@ -216,7 +216,7 @@ This is the behavior if `on_schema_change: ignore`, which is set by default, and
 If you add a column to your incremental model, and execute a `dbt run`, this column will _not_ appear in your target table.
 
-Similarly, if you remove a column from your incremental model, and execute a `dbt run`, this column will _not_ be removed from your target table.
+Similarly, if you remove a column from your incremental model, and execute a `dbt run`, `dbt run` will fail.
```
suggestion: tiny cleanup for conciseness

```suggestion
If you remove a column from your incremental model and execute a `dbt run`, `dbt run` will fail.
```

nghi-ly commented on website/docs/docs/build/incremental-models.md:
```diff
@@ -216,7 +216,7 @@ This is the behavior if `on_schema_change: ignore`, which is set by default, and
 
```
Could you remove "and on older versions of dbt" in the sentence on line 215? This has been the default behavior for a while so is safe to remove. 

I was hoping to make it easier for you by using the built-in Add a suggestion but I can't select that line for some reason 🤷‍♀️

RatandeepSharma replied:
Changes have been implemented as recommended.

nghi-ly reviewed (changes requested):
I chatted with joel today for a technical review. I've reviewed your PR and left some minor feedback. 

Thanks for catching this and letting us know, @RatandeepSharma ! 

Once you make the changes, we can ship it! ❇️

nghi-ly committed:
Merge branch 'current' into patch-1

nghi-ly committed:
Update website/docs/docs/build/incremental-models.md

mirnawong1 commented:
Hey @RatandeepSharma wanted to check in to see if you can make those changes my teammate suggested?

RatandeepSharma committed:
removed "and on older versions of dbt" in the sentence on line 215

Removed "and on older versions of dbt" in the sentence on line 215? This has been the default behavior for a while so is safe to remove.

RatandeepSharma commented:
Hey @mirnawong1 - I've implemented the changes as recommended.

mirnawong1 committed:
Merge branch 'current' into patch-1

mirnawong1 commented on website/docs/docs/build/incremental-models.md:
```diff
@@ -212,11 +212,11 @@ Currently, `on_schema_change` only tracks top-level column changes. It does not
 ### Default behavior
 
-This is the behavior if `on_schema_change: ignore`, which is set by default, and on older versions of dbt.
+This is the behavior if `on_schema_change: ignore`, which is set by default.
```
```suggestion
This is the behavior of `on_schema_change: ignore`, which is set by default.
```

mirnawong1 committed:
Update website/docs/docs/build/incremental-models.md

mirnawong1 reviewed (approved):
thanks again for making those changes and opening up this pr @RatandeepSharma ! great job and merging for you ✨
